File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

//SumMultiples returns the sum of all multiples below a number
func SumMultiples(limit int, factors ...int) int {
	var m = map[int]bool{}
	sum := 0

	for _, factor := range factors {
		if factor != 0 {
			for multiple := factor; multiple < limit; multiple += factor {
				if !m[multiple] {
					sum += multiple
					m[multiple] = true
				}
			}
		}
	}

	return sum
}

File: kotlin/sum-of-multiples/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {
    
    fun sum(factors: Set<Int>, limit: Int): Int {
        var multiples = mutableSetOf<Int>()
        for (factor in factors) {
            for (i in factor..limit - 1 step factor) {
                multiples.add(i)
            }
        }
        return multiples.sum()
    }
}
